### DISCUSSION TITLE
value_to<> method

### DISCUSSION THREAD

**Discussion by jjcasmar** [nlohmann/json#2442] [Q&A] [answered]
Does exist a function that takes a key and a value and return the content of the key into the value?

Would be something like this:

```
json o;
double d = o.value("key", double{});
```

but receiving the placeholder directly, so you dont have to specify the type to which is converted

```
json o;
double d;
o.value_to(key, d);
```
If key doesn't exist, could leave the result as it is or return the return of `decltype(d){}`

**Comment by gregmarr** [comment, marked as answer]
There is one that does the "so you dont have to specify the type to which is converted"

    json o;
    double d;
    o.at(key).get_to(d);

and one which does the "If key doesn't exist, could leave the result as it is or return the return of decltype(d){}"

    json o;
    double d = o.value<double>(key, {});

but no combination of the two as far as I know.  You could do this:

    json o;
    double d = {};

    if (o.find(key) != o.end())
        o.at(key).get_to(d);

and you could probably write your own template to do that:

    template<typename K, typename V>
    json value_to(json const &o, K&& key, V& value) {
        if (o.find(key) != o.end()) {
            o.at(key).get_to(value);
        }
    }
